micro-rdk/src/esp32/entry.rs

```diff
@@ -149,6 +150,9 @@ pub async fn serve_web_inner(
             max_webrtc_connection,
         )
         .with_webrtc(webrtc)
+        .with_periodic_app_client_task(Box::new(RestartMonitor::new(|| unsafe {
+            crate::esp32::esp_idf_svc::sys::esp_restart()
```

Comment:
if there is an easy way I would rather do a unwind of the server before restart so we avoid leaving connections open against app.viam.com

Reply:
Let me think about how that might be achieved. I sense some trickiness about it. Would you be OK with moving forward with this approach for now and tackling clean unwind in a follow-up PR?

Reply:
@npmenard - I'd like to hold off on this unless you feel strongly that we need this piece (which I agree with) before Gautham can integrate.

Reply:
wanted to agree with looking into this (and also that it should be a follow-up for later)